api/Controllers/HotelsController.cs

```
@@ -74,5 +97,69 @@ public async Task<IActionResult> GetHotelCards([FromQuery] PageParameters pagePa
         [HttpGet]
         public async Task<IActionResult> GetHotelsCount() =>
             Ok(await _hotelsDb.Hotels.CountAsync());
+
+        [Route("{hotelId}/images")]
+        [HttpPost]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<IActionResult> AddImage([FromRoute] int hotelId)
+        {
+            try
+            {
+                var file = Request.Form.Files[0];
+                if (file.Length <= 0)
+                {
+                    return BadRequest("Something is wrong with file, probably file is empty");
+                }
+                string fileName = Guid.NewGuid() + Path.GetExtension(file.FileName);
+                string fullPath = Path.Combine(_fileFolder, fileName);
+                using (var stream = new FileStream(fullPath, FileMode.Create))
+                {
+                    await file.CopyToAsync(stream);
+                }
+                ImageEntity image = new ImageEntity
+                {
+                    Path = fileName,
+                    HotelId = hotelId,
+                    Hotel = await _hotelsDb.Hotels
+                        .Where(h => h.Id == hotelId)
+                        .SingleOrDefaultAsync(),
+                    IsOuterLink = false
+                };
+                await _hotelsDb.Images.AddAsync(image);
+                await _hotelsDb.SaveChangesAsync();
+                return CreatedAtAction(nameof(GetImage), new { hotelId = image.Hotel.Id, imageId = image.Id }, null);
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex);
+            }
```

Why catch all exceptions and return 400? 
Is it true for all possible exceptions that they are because of user did something wrong?

---

No it is not. I should also validate images extensions and images size. Gonna fix it in new commit

---

In the next pr))))